### PR TITLE
Fix handling of Debian files with trailing slash

### LINKED ISSUE #26022: deb file upload error
### Description

Hi!
I  can't upload my deb package gitea package registry. Curl request returning "unsupported compression algorithm". Because file names in my deb file ends with slash. 
```
head -100c yonca-scripts_0.0.168_amd64.deb 
!<arch>
debian-binary/  0           0     0     644     4         `
2.0
control.tar.xz/ 0   
```        
I found bellow message on this url https://man7.org/linux/man-pages/man5/deb-split.5.html

       The file is an ar archive with a magic value of !<arch>.  The
       file names might contain a trailing slash (since dpkg 1.15.6). 

### Gitea Version

1.20.0

### Can you reproduce the bug on the Gitea demo site?

Yes

### Log Gist

_No response_

### Screenshots

_No response_

### Git Version

_No response_

### Operating System

debian linux

### How are you running Gitea?

systemd service on linux server

### Database

PostgreSQL

### PR DESCRIPTION
**Pull request by KN4CK3R**
Fixes #26022

- Fix handling of files with trailing slash
- Fix handling of duplicate package file errors
- Added test for both

### CODE REVIEW

File: tests/integration/api_packages_debian_test.go [hunk 1]
```diff
@@ -144,6 +147,14 @@ func TestPackageDebian(t *testing.T) {
...
 								return seen
 							})
+
+							req = NewRequestWithBody(t, "PUT", uploadURL, createArchive(packageName, packageVersion, architecture, false))
+							AddBasicAuthHeader(req, user.Name)
+							MakeRequest(t, req, http.StatusBadRequest)
+
+							req = NewRequestWithBody(t, "PUT", uploadURL, createArchive(packageName, packageVersion, architecture, true))
+							AddBasicAuthHeader(req, user.Name)
+							MakeRequest(t, req, http.StatusBadRequest)
```
Comment (delvh) [inline tests/integration/api_packages_debian_test.go]: Isn't there a test missing for an archive with trailing slashes that succeeds?

Comment (KN4CK3R) [reply tests/integration/api_packages_debian_test.go]: In theory yes but the new test indicates a successfull upload. The duplicate package error is raised after processing the file while inserting it into the database.

**Review by delvh — APPROVED** [review]
I'd still like that test but I can also live with the current status.

**Comment by KN4CK3R** [comment]
I moved the slash testing into the `/modules/packages/debian` unit test.

**Review by lunny — APPROVED** [review]

**Pull request merged**